.github/workflows/scripts/changelog-release-helper.mjs

```diff
@@ -45,16 +45,51 @@ export function validateVersion(newVersion) {
     )
   }
 
-  // Get the version diff keyword (major, minor or patch) which we can use to
-  // help with validating the new version
+  // Get the version diff keyword (major, minor, patch plus prerelease versions
+  // of those 3) which we can use to help with validating the new version
   const versionDiff = semver.diff(newVersion, previousReleaseNumber)
 
-  if (versionDiff === 'major') {
-    checkVersionIncrement(newVersion, previousReleaseNumber, 'major')
-  } else if (versionDiff === 'minor') {
-    checkVersionIncrement(newVersion, previousReleaseNumber, 'minor')
-  } else if (versionDiff === 'patch') {
-    checkVersionIncrement(newVersion, previousReleaseNumber, 'patch')
+  if (!versionDiff) {
+    throw new Error(
+      'Could not determine difference between new and previous versions. Please check the version number you provided and the changelog content.'
+    )
+  }
+
+  // Check if the new version increments from the old version by one for
+  // its change type (major, minor, patch) and throws an error if it doesn't.
+  // Eg: if the current version is 4.3.12:
+  // - 4.3.13, 4.4.0 and 5.0.0 are valid
+  // - 4.3.14, 4.5.0, 6.0.0 and above for all aren't valid
+  let correctIncrement = semver.inc(previousReleaseNumber, versionDiff)
+
+  // We coerce the semvers here to create 2 'plain' semvers in case one of them
+  // is a pre-release eg: 6.1.0-beta.0. This makes it easier to compare the
+  // versions by their major, minor and patch increments.
+  const coercedNewVersion = semver.coerce(newVersion)
+  const coerceCorrectIncrement = semver.coerce(correctIncrement)
+
+  // We check the coerced new version only because there's a chance it could be
+  // null whereas the correct inc we know is a product of semver's processing
+  if (!correctIncrement || !coercedNewVersion || !coerceCorrectIncrement) {
+    throw new Error(
+      'There was a problem comparing the provided version and the most recent version. The format of the provided version may be invalid semver.'
+    )
+  }
+
+  if (!semver.satisfies(coercedNewVersion, `<=${coerceCorrectIncrement}`)) {
+    if (versionIsAPrerelease(newVersion)) {
+      const [identifier, base] = getPrereleaseIdentifierAndBase(newVersion)
+      correctIncrement = semver.inc(
+        correctIncrement,
+        'prerelease',
+        identifier,
+        base
```

Comment:
I _think_ the way it's currently set up, base will always be a '0' or a '1', and any pre-release we do like 6.0.0-beta.20 won't work? We only want to set the base when we do the first pre-release, from then on we want to increment the number.

I can't remember what `base` defaults to, but we've used `'0'` for past releases so I guess we want to continue zero-basing.

Reply:
So the issue here is that this isn't _incrementing_ the version. It's rebuilding what it expects is the correct increment when a version is a pre-release. We strip pre-release away early on for easier comparrisson and then re-establish it later. Obviously that doesn't work if we're not actually comparing pre-release bases properly.

I think I'm going to need to do two things:

1. Figure out how to compare the base of a pre-release, which this currently doesn't do
2. Actually correctly build